docs/source/configuration/external_timelock_service_configs/timelock_server_config.rst

```diff
@@ -131,6 +131,46 @@ default values.
      - The length of time between a follower initiating a ping to a leader and, if it hasn't received a response,
        believing the leader is down, in ms (default: ``5000``).
 
+.. _timelock-server-time-limiting:
+
+Time Limiting
+-------------
+
+Clients that make long-running lock requests will block a thread on TimeLock for the duration of their request. More
+significantly, if these requests are blocked for longer than the idle timeout of the server's application connector
+on HTTP/2, then Jetty will send a stream closed message to the client. This can lead to an infinite buildup of threads
+and was the root cause of issue `#1680 <https://github.com/palantir/atlasdb/issues/1680>`__. We thus reap the thread
+for interruptible requests before the timeout expires, and send an exception to the client indicating that its request
+has timed out, but it is free to retry on the same node. Note that this issue may still occur if a *non-interruptible*
+method blocks for longer than the idle timeout, though we believe this is highly unlikely.
+
+This mechanism can be switched on and off, and the time interval between generating the ``BlockingTimeoutException``
+and the actual idle timeout is configurable. Note that even if we lose the race between generating this exception and
+the idle timeout, we will retry on the same node. Even if this happens 3 times in a row we are fine, since we will fail
```

Comment:
(We spoke about the race condition safety offline. If Jetty wins 3x we'll fail over to the other nodes, which will promptly tell us we're not the leader - so we'll go back to the start!)

Lock fairness - yes, I'll add a note here.